x/ccv/provider/keeper/grpc_query.go

```diff
@@ -437,29 +437,27 @@ func (k Keeper) hasToValidate(
 	if err != nil {
 		return false, nil
 	}
+
+	minPowerToOptIn := int64(0)
+	// If the consumer is TopN compute the minimum power
 	if topN := k.GetTopN(ctx, consumerId); topN > 0 {
-		// in a Top-N chain, we automatically opt in all validators that belong to the top N
-		minPower, found := k.GetMinimumPowerInTopN(ctx, consumerId)
-		if found {
-			k.OptInTopNValidators(ctx, consumerId, activeValidators, minPower)
-		} else {
-			k.Logger(ctx).Error("did not find min power in top N for chain", "chain", consumerId)
+		minPowerToOptIn, err = k.ComputeMinPowerInTopN(ctx, activeValidators, topN)
```

Comment:
Isn't sufficient to try to get the minimum power from the store first and if you cannot find it, then compute it? Can the minimum power in the store be stale?

Reply:
Actually, this is needed as the minimum power stored is indeed stale. The one of the new epoch will be computed at the end of the epoch. Could you add a comment above ComputeMinPowerInTopN?

Reply:
Added.